### PR TITLE
Bookstore: initialize project with components

### PR DESCRIPTION
**Pull request by sadaf-Daneshgar**
**New Features**

- ✅ Setup a new React application using [Create React App](https://reactjs.org/docs/create-a-new-react-app.html#create-react-app).
- ✅ Create a directory for your reusable components: /components.
- ✅ [Setup router](https://reactrouter.com/en/main/start/tutorial#adding-a-router) inside of <App>.
- ✅ Add 2 routes in my application, each should render a page component:
  - `/`
  - `/categories`
- ✅ Create components inside of the /components directory:
  -   Contains individual book state, as well as a button to delete the books.
  - Contains a div to create new books
  - Contains a navigation to link to the 2 routes I created.

### CODE REVIEW

File: src/components/Addbook.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          <button type="button" className="addbtn">
+            ADD BOOK
+          </button>
+        </div>
+      </div>
+    </>
+  );
+}
+
+export default AddBook;
```
Comment (Cabraham1) [inline src/components/Addbook.js]: Great progress on your project so far! 👍🏿 However, in line with the project requirements, you are tasked with developing a book component, which includes a booklist component. This booklist component should receive book data as a prop and display the title and author of each book. Additionally, it should feature a button for removing books.

Please note that the button's functionality for removing books is currently a placeholder and is not yet operational.

<NewBookForm>
This component is responsible for rendering a form containing input fields for both the book's title and author. It also includes a submit button.

Similar to the previous button, the submit button's functionality for adding books is also a placeholder and is not yet functional.

**Review by Cabraham1 — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

## Hey @sadaf-Daneshgar  :wave:

## Highlights :1st_place_medal:

✔️ Descriptive PR Title
✔️ Descriptive PR Summary
✔️ Following some requirements

Good job on implementing some of the requirements, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval but we highly recommend to students that they should take these changes into consideration._

Cheers, and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

---------------------------------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Addbook.js [hunk 2]
```diff
@@ -1,50 +1,108 @@
...
+                Remove
+              </button>
+              <button type="button" className="btn" id="edit">
+                Edit
+              </button>
+            </div>
+            <div className="compelet">
+              <span className="precent">
+                64%
+              </span>
```
Comment (Ibizugbe) [inline src/components/Addbook.js]: Just as the previous reviewer requested, kindly break the `addBook` component into separate components and ensure they are placed in separate files.

for instance, this `AddBook` component can contain the form for adding new books and be responsible for adding new books😉.

then you can create another component `Book` which should be in a separate file `Book.js`. this should take the book data as a prop and render the title and author of a book and has a button for removing books😉.
Lastly, another component called the `bookList` in a `bookList.js` file. this should be responsible for rendering all the books in the array😉.

I understand you have implemented these features in this class, but breaking your project into smaller components is important for component reusability and separation of concerns😉. This makes your codebase readable and maintainable👍🏼.

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @sadaf-Daneshgar 👋🏼👋🏼👋🏼, 

require_changes


Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights✅✅✅

- [x] Linters are passing👍🏼.
- [x] Add and remove functionality works well👍🏼.


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

- [OPTIONAL] Kindly ensure your commit messages are always meaningful ( it reflects the changes implemented) and capitalized as it gives your repo  a more professional look👍🏼.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by usorfaitheloho — APPROVED** [review]
Hi @sadaf-Daneshgar  , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**